dwave/optimization/symbols.pyx

```diff
@@ -2200,6 +2208,42 @@ cdef class LP(Symbol):
             self._objective_value = LPObjectiveValue(self)
         return self._objective_value
 
+    def state(self, Py_ssize_t index = 0):
+        """Return the current solution to the LP.
+
+        If the LP is not feasible, the solution is not meaningful.
+        """
+
+        # While LP is not an ArraySymbol, we nonetheless can access the state
+
+        cdef Py_ssize_t num_states = self.model.states.size()
+        if not -num_states <= index < num_states:
+            raise ValueError(f"index out of range: {index}")
+        elif index < 0:  # allow negative indexing
+            index += num_states
+
+        if not self.model.is_locked() and self.node_ptr.topological_index() < 0:
+            raise TypeError("the state of an intermediate variable cannot be accessed without "
+                            "locking the model first. See model.lock().")
+
+        # Rather than using a StateView, let's just do an explicit copy here
+
+        cdef States states = self.model.states  # for Cython access
+        states.resolve()
+        self.model._graph.recursive_initialize(states._states.at(index), self.node_ptr)
+
+        solution = self.ptr.solution(states._states.at(index))
+
+        cdef double[::1] state = np.empty(self._num_columns(), dtype=np.double)
+
+        if <Py_ssize_t>solution.size() != state.shape[0]:
+            raise RuntimeError  # should never happen, but avoid the segfault just in case
+
+        for i in range(state.shape[0]):
```

Comment:
Might be worth `cdef`ing `i`

Reply:
Cython (usually) does it automatically these days. And I did double check in this case.

Reply:
I thought it might, but I couldn't remember so I looked up their docs and they are still adding a `cdef int i` in their examples!